src/settings.ts

```diff
@@ -165,6 +172,9 @@ export default class Settings {
 	public readonly globstar: boolean = this._getValue(this._options.globstar, true);
 	public readonly ignore: Pattern[] = this._getValue(this._options.ignore, [] as Pattern[]);
 	public readonly markDirectories: boolean = this._getValue(this._options.markDirectories, false);
+	// If 0 or negative maxMatches is given, we revert to infinite matches
+	// eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```

Comment:
can we remove this if I:

* ignore negative and zero `maxMatches`
* introduce a new class method that can bound `maxMatches` between `1` and `Infinity`
* assert `maxMatches` to be gte 1 in the constructor

let me know what's preferred :)

Reply:
IMHO, the first option looks good to me because this logic is already exist for the `deep` and `concurrency` options.